library/core/src/panicking.rs

```diff
@@ -266,16 +277,17 @@ fn assert_failed_inner(
 
     match args {
         Some(args) => panic!(
-            r#"assertion failed: `(left {} right)`
-  left: `{:?}`,
- right: `{:?}`: {}"#,
-            op, left, right, args
+            r#"assertion failed: `({left_name} {op} {right_name})`
+ error: {args},
+  left: `{left_val:?}`,
+ right: `{right_val:?}`
+    at: "#
```

Comment:
This will create messages like
```text
thread 'main' panicked at 'assertion failed: `(1 == 2)`
  left: `1`,
 right: `2`
    at: ', lib/main.rs:2:5
```

It's probably possible to create prettier messages by adding a special case for multi-line messages in the panic runtime here: https://github.com/rust-lang/rust/blob/f2eb9f85b9b52e6538c3c7fc160725963272d471/library/std/src/panicking.rs#L260

Reply:
thx, i fixed this by checking if `msg` has any `'\n'`, and used a different format for those.  I also changed the output format to be more readable.